services/notification/notifiers/status/base.py

```diff
@@ -209,7 +210,32 @@ async def notify(self, comparison: Comparison):
                 payload["url"] = get_pull_url(comparison.pull)
             else:
                 payload["url"] = get_commit_url(comparison.head.commit)
-            return await self.send_notification(comparison, payload)
+
+            base_commit = comparison.base.commit if comparison.base else None
+            head_commit = comparison.head.commit if comparison.head else None
+
+            cache_key = make_hash_sha256(
+                dict(
+                    type="status_check_notification",
+                    repoid=head_commit.repoid,
+                    base_commitid=base_commit.commitid if base_commit else None,
+                    head_commitid=head_commit.commitid if head_commit else None,
+                    notifier_name=self.name,
+                    notifier_title=self.title,
+                )
+            )
+
+            last_payload = cache.get_backend().get(cache_key)
+            if last_payload is NO_VALUE or last_payload != payload:
```

Comment:
if the payload is JSON serializable (not sure if it is), can we use the payload as the key?
It would save us the `last_payload != payload` check... if the payload is there (as a key) we know we sent that before (in the last TTL seconds)

Reply:
I'm not sure that would work as expected.  Imagine the following sequence of events...

1. notify with payload X -> hashes to ABC (not in cache so we send notification)
2. notify with payload X -> hashes to ABC (already cached, skip notification)
3. notify with payload Y -> hashes to DEF (not in cache so we send notification)
4. notify with payload X -> hashes to ABC (already in cache, skip notification)

We need to make sure that the last item there actually DOES send a notification.

Reply:
ah I see... Then indeed my understanding was incorrect.
I thought that case 4 there would indicate a task that got delayed for whatever reason (had to retry for example) and was "stale" data (because something more recent, in this case payload Y) was already delivered.

But indeed I see that case 4 would skip notification and it's not what we want.
Thanks for taking the time to explain.